include/boost/json/object.hpp

```diff
@@ -1166,6 +1166,53 @@ class object
     std::size_t
     erase(string_view key) noexcept;
 
+    /** Erase an element preserving order
+
+        Remove the element pointed to by `pos`, which must
+        be valid and dereferenceable. Thus the @ref end()
+        iterator (which is valid but cannot be dereferenced)
+        cannot be used as a value for `pos`.
```

Comment:
The sentences "Thus the @ref end() iterator (which is valid but cannot be dereferenced) cannot be used as a value for `pos`." is redundant.

If this is expected to be a gotcha (?) I'd move it to a note below.

Reply:
It looks like this is verbatim from https://en.cppreference.com/w/cpp/container/vector/erase. Moved to note section.